packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -3314,28 +3314,23 @@ void main() {
       ),
     );
 
-    await tester.pump();
-
-    final double? abcLabelOpacity = tester.widget<DefaultTextStyle>(
-      find.ancestor(
-        of: find.text('Abc'),
-        matching: find.byType(DefaultTextStyle),
-      ).first,
-    ).style.color?.opacity;
-    final double? bcdLabelOpacity = tester.widget<DefaultTextStyle>(
-      find.ancestor(
-        of: find.text('Bcd'),
-        matching: find.byType(DefaultTextStyle),
-      ).first,
-    ).style.color?.opacity;
-
-    expect(abcLabelOpacity, 1.0);
+    await tester.pumpAndSettle();
 
-    if (bcdLabelOpacity == 1.0) {
-      fail('Bcd''s opacity should be lower than 1.0 since it is disabled');
+    double? defaultTextStyleOpacity(String text) {
+      final String? opacity = tester.widget<DefaultTextStyle>(
+        find.ancestor(
+          of: find.text(text),
+          matching: find.byType(DefaultTextStyle),
+        ).first,
+      ).style.color?.opacity.toStringAsFixed(2);
```

Comment:
If you're having trouble with comparing floating point numbers using the closeTo matcher is preferable:
```dart
expect(defaultTextStyleOpacity('Bcd'), closeTo(0.38, 0.01));
```

Reply:
Thanks for the tip, will update it